### PR TITLE
add new linux rules to nursery

### LINKED ISSUE #868: Linux: Hook library function using dlsym with RTLD_NEXT
https://stackoverflow.com/questions/71695007/hook-library-function-using-custom-method-using-dlsym
https://liveoverflow.com/hooking-on-linux-with-ld_preload-pwn-adventure-3/

`dlsym(-1, ...)`

### LINKED ISSUE #869: linux: get current process information
`/proc/self/exe`
`/proc/self/cmdline`

### PR DESCRIPTION
**Pull request by williballenthin**
closes #868 
closes #869

### CODE REVIEW

File: nursery/get-current-process-command-line.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+rule:
+  meta:
+    name: get current process command line
+    # namespace: host-interation/process
```
Comment (mr-tz) [inline nursery/get-current-process-command-line.yml:4]: why not include the namespaces?

how is this displayed in the default/v/vv output rendering?

File: nursery/get-current-process-command-line.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+rule:
+  meta:
+    name: get current process command line
```
Comment (mr-tz) [inline nursery/get-current-process-command-line.yml:3]: add `on Linux`?

Comment (williballenthin) [reply nursery/get-current-process-command-line.yml:3]: i figured we should split out that naming once we have multiple implementations. thoughts?

**Review by mr-tz — COMMENTED** [review]
logic looks good

File: nursery/get-current-process-command-line.yml (hunk 1 above)
Comment (williballenthin) [reply nursery/get-current-process-command-line.yml:4]:
```suggestion
    namespace: host-interation/process
```

File: nursery/get-current-process-command-line.yml (hunk 2 above)
Comment (mr-tz) [reply nursery/get-current-process-command-line.yml:3]: yeah, that works

File: nursery/get-current-process-file-path.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+rule:
+  meta:
+    name: get current process file path
+    # namespace: host-interation/process
```
Comment (williballenthin) [inline nursery/get-current-process-file-path.yml:4]:
```suggestion
    namespace: host-interation/process
```

File: nursery/hook-routines-via-dlsym-rtld_next.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+rule:
+  meta:
+    name: hook routines via dlsym RTLD_NEXT
+    # namespace: linking/hooking
```
Comment (williballenthin) [inline nursery/hook-routines-via-dlsym-rtld_next.yml:4]:
```suggestion
    namespace: linking/hooking
```

**Pull request merged**